### PR TITLE
Update tna-custodian to work with other projects.

### PR DESCRIPTION
**Pull request by MancunianSam**

### CODE REVIEW

File: .github/workflows/apply.yml [hunk 1]
```diff
@@ -7,52 +7,60 @@ on:
...
+        type: choice
+        options:
+          - tdr
+          - dr2
 permissions:
   id-token: write
   contents: write
 jobs:
+  setup:
+    uses: nationalarchives/dr2-github-actions/.github/workflows/set_terraform_var_names.yml@main
```
Comment (TomJKing) [inline .github/workflows/apply.yml:22]: This is referencing a dr2 repo, does there need to be something to point to the tdr equivalent repo as well? If the repo is a project neutral one, should it be given the name `da-github-actions` instead?

File: .github/workflows/apply.yml [hunk 2]
```diff
@@ -7,52 +7,60 @@ on:
...
+    with:
+      project: ${{ github.event.inputs.project }}
+      environment: ${{ github.event.inputs.environment }}
   apply:
-    uses: nationalarchives/tdr-github-actions/.github/workflows/terraform_apply.yml@main
+    uses: nationalarchives/dr2-github-actions/.github/workflows/terraform_apply.yml@main
```
Comment (TomJKing) [inline .github/workflows/apply.yml:27]: See comments above about referenced repo

File: .github/workflows/check.yml [hunk 3]
```diff
@@ -1,15 +1,16 @@
...
 on:
-  push:
   pull_request:
 permissions:
   id-token: write
   contents: write
 jobs:
   check:
-    uses: nationalarchives/tdr-github-actions/.github/workflows/terraform_check.yml@main
+    uses: nationalarchives/dr2-github-actions/.github/workflows/terraform_check.yml@main
```
Comment (TomJKing) [inline .github/workflows/check.yml:9]: See comments above about referenced repo

File: terraform/root_locals.tf [hunk 4]
```diff
@@ -1,12 +1,12 @@
 locals {
   environment = lower(terraform.workspace)
-  assume_role = local.environment == "mgmt" || local.environment == "sbox" || local.environment == "ddri" ? "arn:aws:iam::${var.tdr_account_number}:role/IAM_Admin_Role" : "arn:aws:iam::${var.tdr_account_number}:role/TDRTerraformRole${title(local.environment)}"
+  assume_role = var.project == "dr2" ? "arn:aws:iam::${var.account_number}:role/${title(local.environment)}TerraformRole" : local.environment == "mgmt" || local.environment == "sbox" || local.environment == "ddri" ? "arn:aws:iam::${var.account_number}:role/IAM_Admin_Role" : "arn:aws:iam::${var.account_number}:role/TDRTerraformRole${title(local.environment)}"
```
Comment (TomJKing) [inline terraform/root_locals.tf]: Referencing a specific TDR IAM role: `TDRTerraformRole ...` is that correct.

**Review by TomJKing — COMMENTED** [review]
Looks good. Thanks for working on this. Mostly just questions, where I've noticed specific TDR/DR2 things.

File: .github/workflows/apply.yml (hunk 1 above)
Comment (MancunianSam) [reply .github/workflows/apply.yml:22]: This is the same as the comment on the other repo. `dr2-github-actions` is a weird mix of this generic workflow and some dr2 specific stuff. I changed it here so as not to break `tdr-github-actions` I've not completely given up on a department wide GitHub actions repo though.

File: terraform/root_locals.tf (hunk 4 above)
Comment (MancunianSam) [reply terraform/root_locals.tf]: There's a change here https://github.com/nationalarchives/da-terraform-configurations/pull/3 that adds the account role which I've used here.

File: .github/workflows/apply.yml (hunk 1 above)
Comment (TomJKing) [reply .github/workflows/apply.yml:22]: Might be worth just popping a note in the README or a comment in the code to explain this.

Comment (MancunianSam) [reply .github/workflows/apply.yml:22]: Makes sense, I've added a note to the README.

**Review by TomJKing — DISMISSED** [review]
Looks good. Thanks

**Review by TomJKing — APPROVED** [review]

**Pull request merged**